engine/src/main/java/org/terasology/engine/rendering/primitives/Tessellator.java

```diff
@@ -157,22 +124,16 @@ public void setLighting(Vector3f v) {
         lighting.set(v);
     }
 
-    public MeshData generateMeshData() {
-        MeshData result = meshData;
-        meshData = new StandardMeshData();
-        return result;
+    public MeshData buildMeshData() {
+        return meshData.clone();
     }
 
     public Mesh generateMesh(ResourceUrn urn) {
         Preconditions.checkNotNull(urn);
-        Mesh result = Assets.generateAsset(urn, meshData, Mesh.class);
-        meshData = new StandardMeshData();
-        return result;
+        return Assets.generateAsset(urn, buildMeshData(), Mesh.class);
```

Comment:
Previous behavior resets Tessellator's state. 
You could reuse tessallator for different objects. you cannot do it now. (needs additional review for usages)

Reply:
There is a clove method in build mesh data

Reply:
But `clone()` not resets `meshData` state... or resets? if yes - then it is not clone. (imho clone should be pure)

Reply:
Umm should I use a different verb. I would need to duplicate the state for the bindings.

Reply:
@pollend @DarkWeird is this resolved?